### PR TITLE
Updates

### PR DESCRIPTION
**Pull request by arnscott**
added self denoising classifier to DPKS and allowed the Loss Interpreter to be used for regression problem

### CODE REVIEW

File: dpks/denoise.py [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+            n_estimators=n_estimators,
+            max_samples=max_samples,
+            bootstrap=True,
+            n_jobs=n_jobs,
+            random_state=random_state,
+        )
+
+    def vote(self, X: np.ndarray, y: np.ndarray, threshold: float = 0.5) -> np.ndarray:
+
+        skf = StratifiedKFold(n_splits=3)
```
Comment (ErikHartman) [inline dpks/denoise.py:48]: Should n_splits maybe be a hyperparameter that can be adjusted?

File: dpks/denoise.py [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            y_fold = y[train]
+
+            self.fit(X_fold, y_fold)
+
+            estimator_probabilities = list()
+
+            for estimator in self.estimators_:
+
+                probabilities = np.where(
+                    estimator.predict_proba(X_fold)[:, 1] >= threshold, 1, 0
```
Comment (ErikHartman) [inline dpks/denoise.py:64]: Why are we predicting on X_fold and not on X[test]?

File: dpks/denoise.py [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        super().__init__(
+            estimator=estimator,
+            n_estimators=n_estimators,
+            max_samples=max_samples,
+            bootstrap=True,
+            n_jobs=n_jobs,
+            random_state=random_state,
+        )
+
+    def vote(self, X: np.ndarray, y: np.ndarray, threshold: float = 0.5) -> np.ndarray:
```
Comment (ErikHartman) [inline dpks/denoise.py:46]: Why do we take a threshold here if 0.5 is always used (theres no way to change the parameters in vote() in predict().

File: dpks/denoise.py [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+        probabilities = np.zeros(
+            (votes.shape[0], 2)
+        )
+
+        probabilities[:, 0] = 1 - votes
+        probabilities[:, 1] = votes
+
+        return probabilities
+
+    def predict(self, X):
```
Comment (ErikHartman) [inline dpks/denoise.py:90]: X: np.ndarray

File: dpks/denoise.py [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+
+            estimator_probability_array = np.array(
+                estimator_probabilities, dtype=np.float64
+            )
+
+            vote_percentages[train] = estimator_probability_array.sum(axis=0) / len(self.estimators_)
+
+        return vote_percentages
+    
+    def predict_proba(self, X) -> np.ndarray:
```
Comment (ErikHartman) [inline dpks/denoise.py:77]: X: np.ndarray

File: dpks/interpretation.py [hunk 6]
```diff
@@ -300,22 +322,22 @@ def fit(self, X, y, classifier) -> None:
...
+
             else:
-                clf.interpret(X_train)
+                clf.fit(X_train, y_train)
+                explainer.fit(clf, X_train)
+
 
             results[f"iteration_{i}_importance"] = pd.Series(
-                clf.mean_importance / clf.mean_importance.max()
+                explainer.global_explanations / explainer.global_explanations.max()
```
Comment (ErikHartman) [inline dpks/interpretation.py:340]: I guess I never saw this before but why do we take .max() here?

**Review by ErikHartman — CHANGES_REQUESTED** [review]
Some comments! Also, tests are failing

File: dpks/interpretation.py (hunk 6 above)
Comment (arnscott) [reply dpks/interpretation.py:340]: This scales all reported importances to the max importance for the column. So they are "normalized" iteration importances. It puts everything between 0-1. The thought was to remove variance based on classifier output just in case the magnitude of the explanations differed between runs for some reason

Comment (ErikHartman) [reply dpks/interpretation.py:354]: Gotcha

**Comment by ErikHartman** [comment]
Approve

**Review by ErikHartman — APPROVED** [review]
